OpenRA.Game/Traits/Player/FrozenActorLayer.cs

```diff
@@ -101,8 +101,9 @@ public void RefreshState()
 				DamageState = health.DamageState;
 			}
 
-			if (tooltip != null)
+			if (tooltips.Any(Exts.IsTraitEnabled))
```

Comment:
You can make this cleaner by pulling a `var tooltip = tooltips.FirstOrDefault(Exts.IsTraitEnabled)` outside the conditional and then keeping this as a null check.